fuzz_test.go

```diff
@@ -1,96 +1,152 @@
-// +build gofuzz
-
 package stun
 
 import (
-	"os"
-	"path/filepath"
-	"strings"
+	"encoding/binary"
+	"errors"
 	"testing"
 )
 
-func TestMessageType_FuzzerCrash1(t *testing.T) {
-	input := []byte("\x9c\xbe\x03")
-	FuzzType(input)
-}
+func FuzzMessage(f *testing.F) {
+	msg1 := New()
+	msg2 := New()
 
-func TestMessageCrash2(t *testing.T) {
-	input := []byte("00\x00\x000000000000000000")
-	FuzzMessage(input)
-}
+	f.Fuzz(func(t *testing.T, data []byte) {
+		msg1.Reset()
+		msg2.Reset()
```

Comment:
Original code reuses only msg1 and recreates msg2
https://github.com/pion/stun/blob/325edd65273f77f871e0b98330babe21974c0999/fuzz.go#L11-L22

Reply:
Yes, I changed this intentionally to speed up the fuzz handler by reducing allocations.

I thought its fine because both messages are reset at the beginning of the handler.
Also to notice, previously the first message was a global variable which I think would break for parallel fuzzing which the Go toolchain uses.

However, I am not sure if the parallel fuzzing is implemented by multi-threading or multi-processing..

WDYT? I can revert the change if there was a reason for reallocating the second message every time :)

Reply:
I'm not very sure about the original objective.
I think comparing reused one and new one may detect problem related to `Reset()`, but comparing two reused ones sounds have less meaning.

> parallel fuzzing

I believe Go never parallelize tests unless `t.Parallel()` is called.

Reply:
> I believe Go never parallelize tests unless t.Parallel() is called.

Thanks, I didnt know this :)

However, while fuzzing tests are executed just like normal unit tests, the actuall fuzzing `go test -fuzz` runs in parallel as I am seeing the following output (".. **with 6 workers**"):

```bash
root@ubuntu-linux-22-04-desktop:~/workspace/pion/stun# go test -fuzz=FuzzMessage
fuzz: elapsed: 0s, gathering baseline coverage: 0/27 completed
fuzz: elapsed: 0s, gathering baseline coverage: 27/27 completed, now fuzzing with 6 workers
fuzz: elapsed: 3s, execs: 235145 (78256/sec), new interesting: 0 (total: 27)
```

To be safe, I reverted the change back to the original.

Reply:
I think it's better for now to keep the original behavior.

But reusing the variable among the calls makes the test nondeterministic depending on the previous random value. So, it would be better to be refactored. (maybe need to check the original objective of the test case)